src/Controller/AnnotationController.php

```diff
@@ -72,7 +87,12 @@ public function feedAction(
         foreach ($annotations as $item) {
             $readerList[$item->getItemId()] = $readerService->getChangeStatus($item->getItemId());
         }
-
+        /**
+         * For first show annotations no read and after mark read.
+         */
+        $date = $this->dateService->getDate($linkedItemId);
+        $annotationList = $date->getAnnotationList();
+        $annotationService->markAnnotationsReadedAndNoticed($annotationList);
```

Comment:
You can not rely on `$linkedItemId` to be of type date here. If you search for `AnnotationController::feedAction` you will see, that this controller action is used in multiple rubrics. 

- Remove the setter injection for DateService
- Get the generic cs_item from ItemService->getItem(...)
- Call getAnnotationList() on that item